lib/puppet/parser/functions/scanf.rb

```diff
@@ -8,27 +10,17 @@
 are similar but not identical to the formats used in Puppet's `sprintf` function.
 
 This function takes two mandatory arguments: the first is the String to convert, and the second
-the format String. A parameterized block may optionally be given, which is called with the result
```

Comment:
Unfortunately the documentation extraction extracts the text for the function from the 3x implementation at the moment. The documentation text must describe both the future parser (4x) version and the 3x version. Suggest rephrasing to language such as "When used with the future parser ..." for the things that are not available on 3x classic.